### PR TITLE
Use custom `states_to_binary` method in `QubitDevice`'s

### PR DESCRIPTION
**Pull request by dwierichs**
This PR changes `QubitDevice.generate_samples` and `QubitDevice.generate_basis_states` to use `self.states_to_binary` instead of `QubitDevice.states_to_binary`.

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -252,6 +252,10 @@
 
 <h3>Bug fixes</h3>
 
+* Fixes a bug where custom implementations of devices' `states_to_binary`
```
Comment (antalszava) [inline doc/releases/changelog-dev.md:255]:
```suggestion
* Fixes a bug where custom implementations of devices' `states_to_binary`
```
```suggestion
* Fixes a bug where the custom implementation of the `states_to_binary` device
```

**Review by antalszava — APPROVED** [review]
Looks good! :tada: Thank you :)

**Comment by dwierichs** [comment]
Hi @antalszava ,thanks for catching the leftover code line :) :facepalm: 
I noticed (based on a codefactor warning) that `QutritDevice` is a child of `QubitDevice` and copied this bug. Added the corresponding changes to `QutritDevice`.

**Review by antalszava — APPROVED** [review]
:clap: Great catch with `QutritDevice`! :100:

**Pull request merged**